pandas/util/_decorators.py

```diff
@@ -265,37 +265,36 @@ def doc(*args: Union[str, Callable], **kwargs: str) -> Callable[[F], F]:
         The string which would be used to format docstring template.
     """
 
-    def decorator(func: F) -> F:
-        @wraps(func)
-        def wrapper(*args, **kwargs) -> Callable:
-            return func(*args, **kwargs)
+    def __init__(self, *args: Union[str, Callable], **kwargs: str) -> None:
+        self.appenders = args
+        self.substitution = kwargs
 
+    def __call__(self, func: F) -> F:
```

Comment:
Does `F` (`FuncType`) also represent classes, or just functions? `func` is now a function or a class (probably worth changing the name).

Reply:
Good point! I agree here. It has been replaced to callable.